implementations/rust/ockam/ockam_abac/src/policy/storage/resource_to_resource_type.rs

```diff
@@ -0,0 +1,21 @@
+use crate::{Resource, ResourceName, ResourceType};
+use ockam_core::async_trait;
+use ockam_core::compat::boxed::Box;
+use ockam_core::Result;
+
+/// This repository stores the relationship between resources and resource types.
+#[async_trait]
+pub trait ResourceToResourceTypeRepository: Send + Sync + 'static {
```

Comment:
One question about the uniqueness check. If we check only the resource name and node, the user won't be able to create an inlet and an outlet with the same name, and that limitation wouldn't make much sense.

Reply:
I thought about that too and:

 - I am not sure it makes sense to use the same name, on the same node (did we ever have an example of this?)
 - It could even be the sign of a configuration error where the same name is reused but shouldn't